src/Compilers/CSharp/Portable/Symbols/MethodSymbolExtensions.cs

```diff
@@ -304,14 +304,25 @@ public static bool IsGenericTaskReturningAsync(this MethodSymbol method, CSharpC
         }
 
         /// <summary>
-        /// Returns whether this method is async and returns an async-enumerable.
+        /// Returns whether this method is async and returns an IAsyncEnumerable`1.
         /// </summary>
-        public static bool IsEnumerableReturningAsync(this MethodSymbol method, CSharpCompilation compilation)
+        public static bool IsIAsyncEnumerableReturningAsync(this MethodSymbol method, CSharpCompilation compilation)
         {
             return method.IsAsync
                 && method.ReturnType.IsIAsyncEnumerableType(compilation);
         }
 
+        public static bool IsBadAsyncReturn(this MethodSymbol method, TypeSymbol returnType)
```

Comment:
It doesn't seem like this function has much to do with methods, aside from the debug assert. Would it be more useful to just have a static method operating on a Compilation and return type?

Reply:
My bad. There shouldn't be a second parameter. We should be using `method.ReturnType` here. I'll fix

---
In reply to: [218948460](https://github.com/dotnet/roslyn/pull/29741#discussion_r218948460) [](ancestors = 218948460)

Reply:
I've reverted this back.
Although I'd forgotten the reason, it's important to pass the return type separately, to avoid cycles with local functions.

Reply:
OK, same question then -- why not just pass a compilation? 😄